### PR TITLE
Hypershift tests

### PR DESCRIPTION
**Pull request by rnetser**
Add tests to install and uninstall a hypershift hosted cluster

### CODE REVIEW

File: tests/hypershift/setup-vpc.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
+variable "aws_region" {
+  type        = string
+  description = "The region to create the ROSA cluster in"
+  default     = "us-east-2"
+
+  validation {
+    condition     = contains(["us-east-2", "us-west-2"], var.aws_region)
```
Comment (ILpinto) [inline tests/hypershift/setup-vpc.tf:7]: If we will need to have other regions ? In stage/prod the check need to support it also

File: tests/hypershift/setup-vpc.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  default = {
+    us-east-2 = ["use2-az1", "use2-az2"]
+    us-west-2 = ["usw2-az1", "usw2-az2"]
+  }
+}
+
+variable "cluster_name" {
+  type        = string
+  description = "The name of the ROSA cluster to create"
+  default     = "rosa-cluster"
```
Comment (ILpinto) [inline tests/hypershift/setup-vpc.tf:27]: msi-rosa-cluster ?

File: tests/hypershift/setup-vpc.tf (hunk 1 above)
Comment (rnetser) [reply tests/hypershift/setup-vpc.tf:7]: moved to python

**Review by myakove — APPROVED** [review]

File: tests/hypershift/test_hypershift.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        raise
+
+
+def verify_exported_aws_credentials():
+    missing_envs = []
+    for var_env in ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]:
+        if not os.getenv(var_env):
+            missing_envs.append(var_env)
+
+    assert not missing_envs, f"Missing environment variables: {missing_envs}"
```
Comment (oharan2) [inline tests/hypershift/test_hypershift.py:51]: I guess the rosa cmd will [catch](https://github.com/RedHatQE/managed-services-integration-tests/pull/55/files#:~:text=if%20%22ERR%22%20in%20cmd_err%3A) configuration errors, but for AWS do we want to consider check and validate the credentials actual values when session starts? same for ROSA, so any of the configuration fails will be raising before further procedures. 
I'm thinking on local envs when aws credentials need to be set and valid.

Comment (rnetser) [reply tests/hypershift/test_hypershift.py:51]: terraform will fail if the credentials are wrong

File: tests/hypershift/test_hypershift.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        "cluster_name": f"msi-{shortuuid.random(11)}".lower(),
+    }
+    LOGGER.info(f"Cluster parameters: {params}")
+    return params
+
+
+@pytest.fixture(scope="class")
+def terraform_workdir(tmp_path_factory):
+    tmp_dir = tmp_path_factory.mktemp("terraform_working_dir")
+    LOGGER.info(f"Terraform work die: {tmp_dir}")
```
[the remark this answers is not in the document]
Comment (rnetser) [reply tests/hypershift/test_hypershift.py:105]: done

File: tests/hypershift/test_hypershift.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    except KeyError:
+        LOGGER.error(
+            f"Failed to get cluster subnets, terraform output: {terraform_output}"
+        )
+        raise
+
+
+@pytest.fixture(scope="class")
+def cluster(ocm_client_scope_session, cluster_parameters):
+    return Cluster(
```
[the remark this answers is not in the document]
Comment (rnetser) [reply tests/hypershift/test_hypershift.py:149]: why?

File: tests/hypershift/test_hypershift.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+    # ROSA output may contain WARNINGS which can be ignored; failing only if there are errors.
+    if "ERR" in cmd_err:
+        LOGGER.error(
+            f"Failed to install cluster: \noutput: {cmd_out}\nerror: {cmd_err}"
+        )
+        raise ClusterInstallError
+
+
+@pytest.fixture(scope="session")
```
Comment (omrirh) [inline tests/hypershift/test_hypershift.py:81]: shouldn't be using a class scope?

File: tests/hypershift/setup-vpc.tf
Comment (myakove) [inline tests/hypershift/setup-vpc.tf:1]: Please move to manifests folder.

**Review by myakove — CHANGES_REQUESTED** [review]

File: tests/hypershift/setup-vpc.tf
Comment (rnetser) [reply tests/hypershift/setup-vpc.tf:1]: done

File: tests/hypershift/test_hypershift.py [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+def rosa_login():
+    env_str = "--env=staging" if py_config["api_server"] == "stage" else ""
+    cmd_succeeded, _, cmd_err = run_command(
+        command=shlex.split(f"rosa login {env_str}")
+    )
+    if not cmd_succeeded:
+        LOGGER.error(f"Failed to login to ROSA on {cmd_err}")
```
Comment (myakove) [inline tests/hypershift/test_hypershift.py:99]:
```suggestion
        LOGGER.error(f"Failed to login to ROSA (env={py_config["api_server"]}) with error: {cmd_err}")
```

**Review by myakove — CHANGES_REQUESTED** [review]

File: tests/hypershift/test_hypershift.py (hunk 7 above)
Comment (rnetser) [reply tests/hypershift/test_hypershift.py]: done

File: README.md [hunk 8]
```diff
@@ -2,51 +2,69 @@
...
+### Prepare a cluster
 
 This project runs tests against an OCP cluster running on OSD / ROSA.
 Export the cluster's kubeconfig file as KUBECONFIG
+
 ```bash
-ocm get clusters -p search="name like '"<cluster name>"%'" | jq -r  '.items | .[] | .id' | xargs -I {} ocm get /api/clusters_mgmt/v1/clusters/{}/credentials | jq -r .kubeconfig > ~/kubeconfig ; export KUBECONFIG=~/kubeconfig
+ocm get clusters -p search="name like '"<cluster name>"%'" | jq -r  '.items | .[] | .id' \
+| xargs -I {} ocm get /api/clusters_mgmt/v1/clusters/{}/credentials | jq -r .kubeconfig > \
+~/kubeconfig ; export KUBECONFIG=~/kubeconfig
```
Comment (myakove) [inline README.md:63]:
```suggestion
export CLUSTER_NAME=<cluster_name>

ocm get clusters -p search="name like '"$CLUSTER_NAME"%'" | jq -r  '.items | .[] | .id' \
| xargs -I {} ocm get /api/clusters_mgmt/v1/clusters/{}/credentials | jq -r .kubeconfig > \
~/kubeconfig ; export KUBECONFIG=~/$CLUSTER_NAME-kubeconfig
```

**Review by myakove — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (rnetser) [reply README.md:63]: This is a matter of preference; I for once prefer not to set `CLUSTER_NAME`

File: builder/Dockerfile
[the remark this answers is not in the document]
Comment (rnetser) [reply builder/Dockerfile:1]: done

File: tests/hypershift/test_hypershift.py [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+
+
+@pytest.fixture(scope="class")
+def cluster_scope_class(ocm_client_scope_session, cluster_parameters):
+    cluster_name = cluster_parameters["cluster_name"]
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=120,
+            sleep=5,
+            func=lambda: Cluster(client=ocm_client_scope_session, name=cluster_name),
```
[the remark this answers is not in the document]
Comment (rnetser) [reply tests/hypershift/test_hypershift.py:201]: The sampler expects a function, should not pass `func=Cluster`

**Comment by omrirh** [comment]
!lgtm

**Review by myakove — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by omrirh** [comment]
needs rebase ? except that lgtm

**Review by myakove — APPROVED** [review]

**Pull request merged**